packages/backend/src/open_payments/payment/outgoing/service.ts

```
@@ -265,10 +268,24 @@ async function createOutgoingPayment(
 
     if (isQuoteError(quoteOrError)) {
       return quoteErrorToOutgoingPaymentError[quoteOrError]
+    } else {
+      quote = quoteOrError
     }
     quoteId = quoteOrError.id
   } else {
     quoteId = options.quoteId
+    if (doExtraChecks) {
+      quote = await deps.quoteService.get({ id: quoteId })
+    }
+  }
+
+  if (doExtraChecks) {
+    if (!quote) {
+      return OutgoingPaymentError.UnknownQuote
+    }
+    if (quote.receiver !== options.grant?.limits?.receiver) {
+      return OutgoingPaymentError.InvalidReceiver
```

In the context of Open Payments, I think we should return a 403 error here (`InsufficientGrant`), since the request itself (and the receiver) is valid, but the client making this outgoing payment request does not have the correct access.

(I'll update the issue to reflect this)

---

got it